AWSCognitoIdentityProvider/AWSCognitoIdentityUser.m

```diff
@@ -203,13 +229,28 @@ -(instancetype) initWithUsername: (NSString *)username pool:(AWSCognitoIdentityU
         self.confirmedStatus = AWSCognitoIdentityUserStatusConfirmed;
 
         NSString * accessTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserAccessToken];
+        NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserIdToken];
+        
+        NSString * idToken = self.pool.keychain[idTokenKey];
         NSString * accessToken = self.pool.keychain[accessTokenKey];
-        //if the session expires > 5 minutes return it and there is at least an accessToken.
-        if(expiration && [expiration compare:[NSDate dateWithTimeIntervalSinceNow:5 * 60]] == NSOrderedDescending && accessToken){
-            NSString * idTokenKey = [self keyChainKey:keyChainNamespace key:AWSCognitoIdentityUserIdToken];
-            AWSCognitoIdentityUserSession * session = [[AWSCognitoIdentityUserSession alloc] initWithIdToken:self.pool.keychain[idTokenKey] accessToken:accessToken refreshToken:refreshToken expirationTime:expiration];
         
-            session.expirationTime = expiration;
+        AWSCognitoIdentityUserSession * session;
+        
+        // Session is available if we have expiration and accessToken.
+        if (expiration && accessToken) {
+            session = [[AWSCognitoIdentityUserSession alloc] initWithIdToken:idToken
+                                                                 accessToken:accessToken
+                                                                refreshToken:refreshToken
+                                                              expirationTime:expiration];
+        }
+        
+        
+        // If the session expires > 2 minutes return it. We need to check both accessToken and id Token expiry
+        // since user can change both of them in Cognito console. We are also checking `expiration` time returned
+        // from Cognito auth result to keep the logic similar to before.
```

Comment:
The comparison of the logic "to before" won't be meaningful after this PR is merged. :) 
```suggestion
        // If the session expires > 2 minutes return it. We need to check both accessToken and id Token expiry
        // since customer can change both of them in Cognito console.
```

Reply:
Changed